clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1968,9 +1972,13 @@ SubscriptionState subscriptions() {
     }
 
     private void maybeUpdateSubscriptionMetadata() {
-        if (subscriptions.hasPatternSubscription()) {
-            updatePatternSubscription(metadata.fetch());
+        if (this.metadataVersionSnapshot < metadata.updateVersion()) {
+            this.metadataVersionSnapshot = metadata.updateVersion();
+            if (subscriptions.hasPatternSubscription()) {
+                updatePatternSubscription(metadata.fetch());
+            }
         }
+        updatePatternSubscription(metadata.fetch());
```

Comment:
I would say this duplicated call ended up here by mistake? we only want to check/update the pattern if the metadata changed and `hasPatternSubscription` (done above on ln 1978)

Reply:
Done